lib/spack/spack/test/cmd/graph.py

```diff
@@ -0,0 +1,76 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack.main import SpackCommand, SpackCommandError
+
+import pytest
+
+
+graph = SpackCommand('graph')
+
+
+def test_graph_ascii(builtin_mock, database):
```

Comment:
If you think it's worth it, we can probably group these into 2 tests:
```python
@pytest.mark.parametrize('arguments',[
    ('--ascii', 'dt-diamond'),
    ...
])
def test_graph_command(builtin_mock, database, arguments):
    """Tests successful usage of the graph command"""
    graph(*arguments)

@pytest.mark.parametrize('arguments',[
    ('--installed', 'dt-diamond'),
    ...
])
def test_graph_command_failures(builtin_mock, database, arguments):
    """Tests usage of the graph command that should result in an error"""
    with pytest.raises(SpackCommandError):
        graph(*arguments)
```

Reply:
To be honest, I find a lot of the pytest logic to be highly unreadable and confusing. Maybe I just don't know pytest well enough, but it has a pretty steep learning curve compared to other unit testing frameworks I've used.

Reply:
@adamjstewart  Nevermind, I can do that in a later PR :smile: